corehq/apps/reports/tasks.py

```diff
@@ -168,23 +170,25 @@ def update_calculated_properties():
             notify_exception(None, message='Domain {} failed on stats calculations with {}'.format(dom, e))
 
 
-def _skip_updating_domain_stats(last_updated=None, last_form_submission=None):
+def get_domains_to_update_es_filter():
     """
-    Skip domain if no forms submitted in the last day
-    AND stats were updated less than a week ago.
-
-    :return: True to skip domain
-     """
-    if not last_updated:
-        return False
-
-    last_updated_ago = datetime.utcnow() - iso_string_to_datetime(last_updated)
-    if last_form_submission:
-        last_form_ago = datetime.utcnow() - iso_string_to_datetime(last_form_submission)
-        new_data = last_form_ago < timedelta(days=1)
-    else:
-        new_data = False
-    return last_updated_ago < timedelta(days=7) and not new_data
+    Returns ES filter to filter domains that are never updated or
+        domains that haven't been updated since a week or domains that
+        have been updated within last week but have new form submissions
+        in the last day.
+    """
+    last_week = datetime.utcnow() - timedelta(days=7)
+    more_than_a_week_ago = filters.date_range('cp_last_updated', lt=last_week)
+    less_than_a_week_ago = filters.date_range('cp_last_updated', gte=last_week)
+    not_updated = filters.missing('cp_last_updated')
+    domains_submitted_today = filters.term('name',
```

Comment:
how may domains is this likely to be?

Reply:
These were around 400 on prod yesterday when I checked.

Reply:
Though, it would be nice to pass `distinct` to the ES query itself, but I haven't found an inbuilt filter for it.

Reply:
Ha, found it https://github.com/dimagi/commcare-hq/pull/23727/commits/d4ffb6577b86eda2e59e68702973f078b20c4248